Documentation/Testing.md

```diff
@@ -0,0 +1,160 @@
+# Testing Ladybird
+
+Tests are locates in `Tests/`, with a directory for each library.
+
+Every feature or bug fix added to LibWeb should have a corresponding test in `Tests/LibWeb`.
+The test should be either a Text, Layout, Ref, or Screenshot test depending on the feature.
+Tests of internal C++ code go in their own `TestFoo.cpp` file in `Tests/LibWeb`.
+
+## Running Tests
+
+> [!NOTE]  
+> To reproduce a CI failure, see the section on [Running with Sanitizers](#running-with-sanitizers).
+
+The easiest way to run tests is to use the `ladybird.sh` script. The LibWeb tests are registered with CMake as a test in
+`Ladybird/CMakeLists.txt`. Using the built-in test filtering, you can run all tests with `Meta/ladybird.sh test` or run
+just the LibWeb tests with `Meta/ladybird.sh test LibWeb`. The second way is to invoke the headless browser test runner
+directly. See the invocation in `Ladybird/CMakeLists.txt` for the expected command line arguments.
+
+A third way is to invoke `ctest` directly. The simplest method is to use the `default` preset from ``CMakePresets.json``:
+
+```sh
+cmake --preset default
+cmake --build --preset default
+ctest --preset default
+```
+
+If you want to avoid building and running LibWeb tests, you can use a Lagom-only build.
```

Comment:
We can probably yeet this part. I'm not sure if building with Lagom as source directory works anymore, it's not tested on CI.